### PR TITLE
 #126 create own method for every entity type to prevent java function limit of 65536 bytes

### PR DESCRIPTION
**Pull request by ManuelB**
This is for bug #126

### CODE REVIEW

**Comment by csadilek** [comment]
Thanks, Manuel. I've looked into this now. I think it's a good start but doesn't fully solve the method byte limit problem yet. You have moved the creation of the anonymous inner classes (entity types) to newly generated private methods. However, the code of these anonymous inner classes didn't count towards the method limit of `createMetaModel` (you can verify this using javap) as inner classes are compiled into separate files. To address this I would suggest to also move the corresponding `addAttribute` calls to your newly generated methods. What do you think?

File: errai-jpa/errai-jpa-client/src/main/java/org/jboss/errai/jpa/rebind/ErraiEntityManagerGenerator.java [hunk 1]
```diff
@@ -241,18 +241,23 @@ public static EntityManagerFactory createHibernateEntityManagerFactory(TreeLogge
...
-  private String generateErraiEntityType(final EntityType<?> et, MethodBlockBuilder<?> pmm, List<MetaClass> globalListeners) {
+  private String generateErraiEntityType(final EntityType<?> et, MethodBlockBuilder<?> pmm, List<MetaClass> globalListeners, ClassStructureBuilder<?> classBuilder) {
     MetaClass met = MetaClassFactory.get(et.getJavaType());
     pmm.append(Stmt.codeComment(
         "**\n" +
         "** EntityType for " + et.getJavaType().getName() + "\n" +
         "**"));
     String entityTypeVarName = entitySnapshotVarName(et.getJavaType());
+    
+    String entityTypeMethodName = "createEntityType"+et.getJavaType().getName().replace('.', '_');
```
Comment (csadilek) [inline errai-jpa/errai-jpa-client/src/main/java/org/jboss/errai/jpa/rebind/ErraiEntityManagerGenerator.java]: The name of this new method should prob. be "createEntityType_" + .... for easier readability: createEntityTypeorg_jboss_errai_demo_client_shared_UserComplaint() vs. createEntityType_org_jboss_errai_demo_client_shared_UserComplaint()

**Comment by ManuelB** [comment]
Hi Christian,
go ahead and do all necessary changes. I can just say that my program is compiling. So it seems that my changes influenced the size of the method :-)

Is it possible to release this in the 3.1.2 branch as well in 3.1.3?

/Manuel

**Comment by csadilek** [comment]
Yeah, but it doesn't actually fix the issue as you or someone else will likely run into this having more entities on the classpath. BTW, do you intend to use JPA on the client-side (actually persist some of your entities in the browser's offline storage)? Because if you don't, you do not need this file at all and you're needlessly increasing your JS file size.

**Comment by ManuelB** [comment]
Hi Christian,
actually this is the number one reason why we are choosing Errai. Currently the problem is gone. If it comes again I will fix it again. I would love to spend time enhancing errai but I have to concentrate on business logic.

/Manuel

**Pull request merged**